src/basic/namespace-util.c

```diff
@@ -625,3 +637,127 @@ int is_idmapping_supported(const char *path) {
 
         return true;
 }
+
+int userns_get_base_uid(int userns_fd, uid_t *ret_uid, gid_t *ret_gid) {
```

Comment:
IIUC this is just a special case of userns_load_range_by_fd? Can the common part of process spawning be split out?

Reply:
i started out like that, but gave up, the differences are too big, because load_range_by_fd()'s base uid mgmt doesn't work for this at all: it always only loads the osurce or the target base, not both.

it's hard to change that. it might be worth doing that, that's something we can discuss, but that's really no material for this PR I am sure. hope that makes sense.